blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -231,6 +233,22 @@ public Link getLink() {
         };
     }
 
+    @Override
+    public List<Object> getParameters() {
+        return getParameterDefinitions(job);
+    }
+
+    public static List<Object> getParameterDefinitions(Job job){
```

Comment:
https://github.com/jenkinsci/pipeline-model-definition-plugin is where it lives. There are NOT IDs, but Andrew came up with a way to refer to specific nodes within the pipeline by a path identifier, which we could use... e.g. `/pipeline/stages/1/branches/0/...`